### PR TITLE
Creation of a PredictManager from MapsManager

### PR DESCRIPTION
**Pull request by camillebrianceau**
Le but de cette PR est d'extraire un Predict/Interpret Manager du MapsManager pour le diviser en plusieurs classe séparées et y voir plus clair 

high level:
- interpret
- predict (from which I extract predict_multi and predict_single)

low level:
- _write_data_group
- _check_leakage
- get_group_info
- _chack_data_group
- _compute_output_nifti
- _compute_latent_tensor

method that are never used:
- get_interpretation (in a test)
- get_model
- get_best_epoch 
- meta_maps_analysis -> get_metrics ?

### CODE REVIEW

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 1]
```diff
@@ -1954,76 +1439,13 @@ def _compute_output_tensors(
...
-                    )
-                latent = latent.squeeze(0).cpu().float()
-                participant_id = data["participant_id"]
-                session_id = data["session_id"]
-                mode_id = data[f"{self.mode}_id"]
-                output_filename = (
-                    f"{participant_id}_{session_id}_{self.mode}-{mode_id}_latent.pt"
-                )
-                torch.save(latent, tensor_path / output_filename)
+    def _find_splits(self):
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py:1438]: `_find_splits` is never used in the trainer part. Is there a reason why you kept it in MAPSManager?

**Review by thibaultdvx — COMMENTED** [review]
LGTM. Only a small question on `_find_splits`

File: clinicadl/utils/maps_manager/maps_manager.py (hunk 1 above)
Comment (camillebrianceau) [reply clinicadl/utils/maps_manager/maps_manager.py:1438]: It is used in the resume part somwhere !

**Review by thibaultdvx — APPROVED** [review]
LGTM

**Pull request merged**